dashboard/app/models/script.rb

```diff
@@ -177,6 +177,33 @@ def self.artist_script
     Script.get_from_cache(Script::ARTIST_NAME)
   end
 
+  # Get the set of scripts that are valid for the current user, ignoring those
+  # that are hidden based on the user's permission.
+  # @param [User] user
+  # @return [AssignableInfo[]]
+  def self.valid_scripts(user)
```

Comment:
I _think_ I recreated this correctly, but my confidence is low.  Please help verify!